### PR TITLE
Tests configs

### PR DESCRIPTION
**Pull request by GustavoFSProg**
This PR is to do some configs in testes and du some testes

### CODE REVIEW

**Comment by GustavoFSProg** [comment]
I made the tests basics, for the components e and page Home, testing if irts render the component and if execute de styles config. By so far all is working. I change a litle parts of the code and intall react-scripts to run the tests, because its the more ezsy way to run the tests script, And its working fine.

File: package.json [hunk 1]
```diff
@@ -6,13 +6,16 @@
...
-    "test": "react-scripts test --no-cache --silent",
+    "test": "react-scripts test --no-cache --silent --watchAll",
     "lint": "./node_modules/.bin/eslint './**/*.{js,ts,tsx}' --quiet"
   },
   "dependencies": {
+    "@types/react-test-renderer": "^17.0.1",
     "@vitejs/plugin-react": "^1.0.7",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-scripts": "^5.0.0",
```
Comment (roavellarm) [inline package.json]: Remove react-scripts, we are using Vite instead

File: package.json [hunk 2]
```diff
@@ -6,13 +6,16 @@
     "dev": "vite",
     "build": "tsc && vite build",
     "preview": "vite preview",
-    "test": "react-scripts test --no-cache --silent",
+    "test": "react-scripts test --no-cache --silent --watchAll",
     "lint": "./node_modules/.bin/eslint './**/*.{js,ts,tsx}' --quiet"
   },
   "dependencies": {
+    "@types/react-test-renderer": "^17.0.1",
```
Comment (roavellarm) [inline package.json]: Remove @types/react-test-renderer

File: package.json [hunk 3]
```diff
@@ -6,13 +6,16 @@
     "dev": "vite",
     "build": "tsc && vite build",
     "preview": "vite preview",
-    "test": "react-scripts test --no-cache --silent",
+    "test": "react-scripts test --no-cache --silent --watchAll",
```
Comment (roavellarm) [inline package.json]: Use jest instead of react-scripts, like this:
```JSON
{
  "test": "jest --no-cache --silent"
}
```

File: package.json (hunk 1 above)
Comment (roavellarm) [reply package.json]: Also, please add a coverage script so we can check it eventually:
```JSON
{
  "coverage": "jest --coverage --watchAll"
}
```

File: package.json [hunk 4]
```diff
@@ -6,13 +6,16 @@
...
     "lint": "./node_modules/.bin/eslint './**/*.{js,ts,tsx}' --quiet"
   },
   "dependencies": {
+    "@types/react-test-renderer": "^17.0.1",
     "@vitejs/plugin-react": "^1.0.7",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-scripts": "^5.0.0",
+    "react-test-renderer": "^17.0.2",
```
Comment (roavellarm) [inline package.json]: Remove react-test-renderer

File: package.json [hunk 5]
```diff
@@ -6,13 +6,16 @@
...
     "styled-components": "^5.3.3",
     "vite": "^2.8.0"
   },
```
Comment (roavellarm) [inline package.json]: Add two new DEV dependencies:
```BASH
$ yarn add -D ts-node identity-obj-proxy
```

**Review by roavellarm — COMMENTED** [review]
After all the dependencies updates I've suggested in the code, do the following:

1. Add coverage after dist-ssr (line 12) in the .gitignore file.
2. Create a `jest.config.ts` file at the root directory of the project and add this:
```JS
export default {
  testEnvironment: 'jsdom',
  transform: {
    '^.+\\.tsx?$': 'ts-jest',
  },
  moduleNameMapper: {
    '\\.(gif|ttf|eot|svg|png)$': '<rootDir>/test/__mocks__/fileMock.js',
    '\\.(css|less|sass|scss)$': 'identity-obj-proxy',
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
}
```
3. Create a `jest.setup.ts` file at the root directory of the project and add this:
```JS
import '@testing-library/jest-dom/extend-expect' // eslint-disable-line
```
4. Create a "test" directory on the root, inside it a "__mock__" directory with a `fileMock.js` file it. So it would be like: `test/__mocks__/fileMock.js`. Copy the following to the file:
```JS
module.exports = 'test-file-stub'
```
5. Do a local backup of all other files you have changed and remove them from this PR. We will separate each concern in a specific branch/PR

**Review by roavellarm — CHANGES_REQUESTED** [review]
Don't forget to remove all the files from this PR.

Do a local backup of them before, so you don't lose your work that we'll just separate it in other PR after we talk about it.

The files to be removed are:

- src/Home/Home.tsx 
- src/Home/style_home.ts 
- src/__test__/Footer.test.tsx 
- src/__test__/Header.test.tsx 
- src/__test__/Home.test.tsx 
- src/__test__/HomeText.test.tsx 
- src/components/Footer/Footer.tsx 
- src/components/Footer/style_footer.ts 
- src/components/Header/header.tsx 
- src/components/Header/style_header.ts 
- src/components/HomeText/HomeText.tsx 
- src/components/HomeText/style_homeText.ts 
- src/styles/styled-app.ts

**Comment by GustavoFSProg** [comment]
I do the review requests, deleting old files and do the configs in jest,config.ts, jest.setup.ts, in package json, ins the sripts f testing. And remove react-scripts, and reat-test-renderer.  I upload the new files configurations.

File: src/__test__/App.test.tsx [hunk 6]
```diff
@@ -1,9 +0,0 @@
-import { render } from '@testing-library/react'
-
-import { App } from '../App'
-
-describe('App', () => {
-  it('should render correctly', () => {
-    render(<App />)
-  })
-})
```
Comment (roavellarm) [inline src/__test__/App.test.tsx]: Hey @GustavoFSProg,  

Nice work!
Please, can you add this test back and test that the App is rendering the `<h1>Hello World</h1>`?
Here is an example:
```JS
it('should render correctly', () => {
    render(<App />)
    expect(screen.getByRole('heading', { name: /Hello World/i })).toBeInTheDocument()
    // OR use Truthy
    expect(screen.getByRole('heading', { name: /Hello World/i })).toBeTruthy()
})
```

File: src/styles/theme.ts [hunk 7]
```diff
@@ -8,7 +8,7 @@ export const theme = {
 
   fontFamily: {
     oswald: 'Oswald, sans-serif',
-    raleway: 'Raleway, sans-serif',
```
Comment (roavellarm) [inline src/styles/theme.ts]: Please, roll back this space, and in a new PR will do next you can change on the test instead of the theme.ts file.

**Review by roavellarm — APPROVED** [review]
Nice work! :rocket:

**Pull request merged**